Sage/carbon
# fix(popover-container): ensure styling is correct when rendered as a child of menu

fix #6538

### Proposed behaviour


Re-introduces Portal to component so that is is not a direct descendant of any potential component
that may cascade styles targeting the container's children elements.

Adds `useFocusPortalContent` hook to handle overriding the tab ordering

![image](https://github.com/Sage/carbon/assets/44157880/f411ba44-e02b-4ed8-9507-0b05e84ec8b0)

### Current behaviour


Container is not rendered in Portal meaning any children may receive style overrides form any parent elements

![image](https://github.com/Sage/carbon/assets/44157880/13a1df79-36ee-440d-b5b3-92dd3d3ebc07)

### Checklist




#### QA


### Additional context



### Testing instructions



The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

**For QA:** I've added a new test story to check that the issue raised is fixed as part of the proposed changes
